src/login/msal/PublicClientCredential.ts

```diff
@@ -19,8 +19,12 @@ export class PublicClientCredential implements TokenCredential, AzExtServiceClie
 		this.accountInfo = accountInfo;
 	}
 
-	public async getToken(scopes: string | string[]): Promise<AccessToken | null> {
-		scopes = Array.isArray(scopes) ? scopes : [scopes];
+	public async getToken(scopes?: string | string[]): Promise<AccessToken | null> {
+		if (scopes) {
+			scopes = Array.isArray(scopes) ? scopes : [scopes];
+		} else {
+			scopes = [];
```

Comment:
If no scopes are passed in, do we want to use `defaultMsalScopes` instead of an empty array?
```suggestion
			scopes = defaultMsalScopes;
```
A token with no scopes is useless right? At least one with the default has a good chance of working (right?).

Reply:
Believe it or not providing no scopes actually gives you with a functional access token 🤷‍♂️ 

For instance in my testing with the ML extension I was able to list all resources in the tree view with an access token I obtained with `scopes = []`.

Reply:
*sigh* of course it works. Uncut key opens all locks.
![image](https://user-images.githubusercontent.com/36966225/154731897-094fdc23-8031-484b-b288-9c56572cf233.png)

🤷‍♂️